libraries/ms-drivers/test/test_spv1020_mppt.c

```diff
@@ -0,0 +1,136 @@
+#include "log.h"
+#include "spv1020_mppt.h"
+#include "spv1020_mppt_defs.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_SPI_PORT SPI_PORT_2
+
+#define TEST_BAUDRATE 60000
+#define TEST_MOSI_PIN \
+  { .port = GPIO_PORT_B, 15 }
+#define TEST_MISO_PIN \
+  { .port = GPIO_PORT_B, 14 }
+#define TEST_SCLK_PIN \
+  { .port = GPIO_PORT_B, 13 }
+#define TEST_CS_PIN \
+  { .port = GPIO_PORT_B, 12 }
+
+void setup_test() {
+  gpio_init();
+  SpiSettings spi_settings = {
+    .baudrate = 60000,
+    .mode = SPI_MODE_0,
+    .mosi = TEST_MOSI_PIN,
+    .miso = TEST_MISO_PIN,
+    .sclk = TEST_SCLK_PIN,
+    .cs = TEST_CS_PIN,
+  };
+  spi_init(TEST_SPI_PORT, &spi_settings);
+}
+void teardown_test() {}
+
+// Test that we can successfully send the SHUT and Turn ON commands.
+void test_shut_and_turn_on(void) {
+  TEST_ASSERT_OK(spv1020_shut(TEST_SPI_PORT));
+  TEST_ASSERT_OK(spv1020_turn_on(TEST_SPI_PORT));
+  TEST_ASSERT_OK(spv1020_shut(TEST_SPI_PORT));
+  TEST_ASSERT_OK(spv1020_shut(TEST_SPI_PORT));  // 2 in a row
```

Comment:
Wait if it is already shut why are they allowed to keep sending this command?

Reply:
I suppose they aren't supposed to, but since the module doesn't keep state I guess this test is to make sure nothing funky happens if it's called multiple times. I'll add a warning when shut is called when it's already shut on x86.